### PR TITLE
CR-1126348 display Config Mode & Max Power when these are non zero values and fix mechanical error info in xbmgmt/xbutil examine reports

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

#### Problem solved by the commit
1. On versal platform, VMR doesn't send Config Mode and Max Power information. So, XRT should skip printing these information as part of xbmgmt examine reports.
2. sdm_sensor_info query request throws proper error if it is unable to find required sysfs path instead of returning empty output.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
CR-1126348 and CR-1121356

#### How problem was solved, alternative solutions (if any) and why they were rejected
1. Added change in XRT UI to display Config Mode & Max Power data only when it is non zero.
2. Mechanical report in xbmgmt/xbutil examine output in json formats. It shows "Information unavailable" if VMR doesn't send required sensor data on versal.

#### Risks (if any) associated the changes in the commit
NA

#### What has been tested and how, request additional testing if necessary
xbmgmt & xbutil examine reports.

#### Documentation impact (if any)
NA

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
Output looks like below on versal, there is no entry for "Config Mode".

`-bash-4.2# xbmgmt examine -r platform -d

-------------------------------------------------------
1/1 [0000:d8:00.0] : xilinx_vck5000_gen4x8_xdma_base_1
-------------------------------------------------------
Flash properties
  Type                 : ospi_xgq
  Serial Number        : XFL1TQYAVK2A

Device properties
  Type                 : vck5000
  Name                 : VCK5000-PP
  Max Power            : N/A

Flashable partitions running on FPGA
  Platform             : xilinx_vck5000_gen4x8_xdma_base_1
  SC Version           : 52.52.51
  Platform UUID        : BEA8EDE9-F61A-CAA3-734D-2184E1C9D1D2
  Interface UUID       : 0EDF299D-075D-87E2-A086-28F732576A3D

Flashable partitions installed in system
  Platform             : xilinx_vck5000_gen4x8_xdma_base_1
  SC Version           : 4.4.32
  Platform UUID        : BEA8EDE9-F61A-CAA3-734D-2184E1C9D1D2

Bootable Partitions:
  Default              : INACTIVE
  Backup               : ACTIVE


  Mac Address          : 00:0A:35:0C:F6:3C:00:A4
                       : 00:0A:35:0C:F6:3D:00:00

WARNING  : SC image on the device is not up-to-date.
`

**Comment by rajkumar-xilinx** [comment]
In JSON report following will be displayed
`            "mechanical":
            {
                "fans":
                {
                    "error_msg": "Information unavailable"
                }
            },
`

**Comment by rajkumar-xilinx** [comment]
Attached complete json output file below

[xbmgmt_examine_json_output.txt](https://github.com/Xilinx/XRT/files/8351371/xbmgmt_examine_json_output.txt)
[xbutil_examine_json_output.txt](https://github.com/Xilinx/XRT/files/8351372/xbutil_examine_json_output.txt)

**Review by stsoe — CHANGES_REQUESTED** [review]
For 2022.1 we stick with Xilinx copright.
But you are welcome to add the AMD one on a separate line as:
Copyright (C) 2022 Advanced Micro Devices, Inc. All rights reserved.

File: src/runtime_src/core/common/sensor.cpp [hunk 1]
```diff
@@ -465,8 +468,14 @@ read_thermals(const xrt_core::device * device)
...
     output = xrt_core::device_query<xq::sdm_sensor_info>(device, xq::sdm_sensor_info::sdr_req_type::thermal);
-    if (output.empty())
-      is_data_driven = false;
+    if (output.empty()) {
+      thermal_array.put("error_msg", "Information unavailable");
+      root.add_child("thermals", thermal_array);
+      return root;
+    }
+  }
+  catch(const xrt_core::query::sysfs_error&) {
```
Comment (AShivangi) [inline src/runtime_src/core/common/sensor.cpp:477]:
```suggestion
  catch(const xrt_core::query::exception&) {
```

File: src/runtime_src/core/common/sensor.cpp [hunk 2]
```diff
@@ -494,8 +503,14 @@ read_mechanical(const xrt_core::device * device)
...
     output = xrt_core::device_query<xq::sdm_sensor_info>(device, xq::sdm_sensor_info::sdr_req_type::mechanical);
-    if (output.empty())
-      is_data_driven = false;
+    if (output.empty()) {
+      fan_array.put("error_msg", "Information unavailable");
+      root.add_child("fans", fan_array);
+      return root;
+    }
+  }
+  catch(const xrt_core::query::sysfs_error&) {
```
Comment (AShivangi) [inline src/runtime_src/core/common/sensor.cpp:512]:
```suggestion
  catch(const xrt_core::query::exception&) {
```

**Review by AShivangi — CHANGES_REQUESTED** [review]
minor changes requested

**Comment by dayeh-xilinx** [comment]
retest this please.

**Review by stsoe — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]

**Comment by rajkumar-xilinx** [comment]
Verified this change on U50 also. 
`
[root@localhost build]# xbmgmt examine -r platform -d 0000:04:00.0

-----
1/1 [0000:04:00.0] : xilinx_u50_gen3x16_xdma_201920_3
-----
Flash properties
  Type                 : spi
  Serial Number        : 50121119CG3R

Device properties
  Type                 : u50
  Name                 : ALVEO U50 PQ
  Config Mode          : 0x7
  Max Power            : 75W

Flashable partitions running on FPGA
  Platform             : xilinx_u50_gen3x16_xdma_201920_3
  SC Version           : 5.2.16
  Platform UUID        : F465B0A3-AE8C-64F6-19BC-150384ACE69B
  Interface UUID       : 862C7020-A250-293E-3203-6F19956669E5

Flashable partitions installed in system
  Platform             : xilinx_u50_gen3x16_xdma_base_5
  SC Version           : 5.2.16
  Platform UUID        : 44654095-25B4-C06A-EC6D-0B479D3FEBE8

  Platform             : xilinx_u50_gen3x16_xdma_201920_3
  SC Version           : 5.2.16
  Platform UUID        : F465B0A3-AE8C-64F6-19BC-150384ACE69B


  Mac Address          : 00:0A:35:06:63:38
                       : 00:0A:35:06:63:39
                       : 00:0A:35:06:63:3A
                       : 00:0A:35:06:63:3B

WARNING  : Multiple shells are installed on the system.

`

File: src/runtime_src/core/pcie/linux/device_linux.cpp [hunk 3]
```diff
@@ -266,7 +266,7 @@ struct sdm_sensor_info
     }
 
     if (path.empty())
-      return result_type();
+      throw xrt_core::query::sysfs_error("sysfs path not found");
```
Comment (rozumx) [inline src/runtime_src/core/pcie/linux/device_linux.cpp:269]: Could you help the future developer an let us know "what path" was not found.   

I ask for this exception basically states....I found an issue, but I am not going to tell you what it is.

**Review by rozumx — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/pcie/linux/device_linux.cpp (hunk 3 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/pcie/linux/device_linux.cpp:269]: Hi @rozumx please let me know if below error message is ok here, thank you.

> throw xrt_core::query::sysfs_error("target hwmon sysfs path /sys/bus/pci/devices/<bdf>/hwmon/hwmon*/ not found");

Comment (rozumx) [reply src/runtime_src/core/pcie/linux/device_linux.cpp:269]: This will work.

**Review by rozumx — CHANGES_REQUESTED** [review]

**Comment by rajkumar-xilinx** [comment]
retest this please

**Review by rozumx — APPROVED** [review]

**Pull request merged**